HomeWorkLinq/HomeWorkLinq/Program.cs

```diff
@@ -0,0 +1,145 @@
+                            new Actor() { Name = "Damon Damon", Birthdate = new DateTime(1970, 8, 10)},
+                            new Actor() { Name = "Robin Williams", Birthdate = new DateTime(1951, 8, 11)},
+                        }},
+                        new Book() { Author = "Stephen King", Name="Finders Keepers", Pages = 200},
+                        "Leonardo DiCaprio"
+                    };
+
+            //1. Output all elements excepting ArtObjects
+            var allElementsWithoutArtObjects = data.Except(data.Where(_ => _ is ArtObject)).ToList();
+
+            //2.Output all actors names
+            var allActorsNames = data.Where(_ => _ is Film)
+                                     .SelectMany(_ => (_ as Film).Actors)
+                                     .Select(actor => actor.Name)
+                                     .ToList();
+
+            //3.Output number of actors born in august
+            var actorsBornInAugust = data.Where(_ => _ is Film)
+                                         .SelectMany(_ => (_ as Film).Actors)
+                                         .Where(actor => actor.Birthdate.Month == 8)
+                                         .Select(actor => actor)
+                                         .Count();
+
+            //4.Output two oldest actors names
+            var twoOldestActorsNames = data.Where(_ => _ is Film)
+                                           .SelectMany(_ => (_ as Film).Actors)
+                                           .OrderBy(actor => actor.Birthdate)
+                                           .Take(2)
+                                           .Select(actor => actor.Name)
+                                           .ToList();
+
+            //5.Output number of books per authors
+            var numberOfBooksPerAuthors = data.Where(_ => _ is Book)
+                                              .GroupBy(_ => (_ as Book).Author)
+                                              .Select(_ => new { NameAuthor = _.Key, CountOfBooks = _.Count() })
+                                              .ToList();
+
+            //6.Output number of books per authors and films per director
+            var numberOfBooksPerAuthorsAndFilmsPerDirector
+                        = data.Where(_ => _ is Book || _ is Film)
+                              .GroupBy(_ => _.GetType())
+                              .Select(_ => new { Author = _.Key, CountArts = _.Count() })
+                              .ToList();
+
+            //7.Output how many different letters used in all actors names
+            var howManyDifferentLettersInNames = data.Where(_ => _ is Film)
+                                     .SelectMany(_ => (_ as Film).Actors)
+                                     .SelectMany(actor => actor.Name)
+                                     .Distinct()
+                                     .Where(_ => _ != ' ')   // Is the space character a symbol?
```

Comment:
Space is a symbol, but not a letter, so excluding space as you did.

Reply:
OK